components/script/dom/headers.rs

```diff
@@ -93,19 +94,19 @@ impl HeadersMethods for Headers {
         let (mut valid_name, valid_value) = try!(validate_name_and_value(name, value));
         valid_name = valid_name.to_lowercase();
         // Step 3
-        if self.guard == Guard::Immutable {
+        if *self.guard.borrow() == Guard::Immutable {
             return Err(Error::Type("Guard is immutable".to_string()));
         }
         // Step 4
-        if self.guard == Guard::Request && is_forbidden_header_name(&valid_name) {
+        if *self.guard.borrow() == Guard::Request && is_forbidden_header_name(&valid_name) {
             return Ok(());
         }
         // Step 5
-        if self.guard == Guard::RequestNoCors && !is_cors_safelisted_request_header(&valid_name) {
+        if *self.guard.borrow() == Guard::RequestNoCors && !is_cors_safelisted_request_header(&valid_name) {
             return Ok(());
         }
         // Step 6
-        if self.guard == Guard::Response && is_forbidden_response_header(&valid_name) {
+        if *self.guard.borrow() == Guard::Response && is_forbidden_response_header(&valid_name) {
```

Comment:
This method (`Append`) here and `Delete`, `Set` could really use a `match` expression instead of using so many `if`s. Something like the following:
```rust
match *self.guard.borrow() {
    Guard::Immutable => Err(Error::Type("Guard is immutable".to_owned())),
    Guard::Request if is_forbidden_header_name(&valid_name) => Ok(()),
    // ... and so on...
}
```